runtime/tests/test_tensor.cpp

```diff
@@ -0,0 +1,42 @@
+#include <cstdlib>
+#include <iostream>
+#include "blueoil.hpp"
+
+int test_tensor() {
+    blueoil::Tensor tensor= blueoil::Tensor::zeros({2, 3});
+    tensor.data[0] = 1;
+    tensor.data[1] = 2;
+    tensor.data[2] = 3;
+    tensor.data[3] = 7;
+    tensor.data[4] = 8;
+    tensor.data[5] = 9;
+    float expected_data[][3] = {
+				{1, 2, 3},
+				{7, 8, 9}
+    };
+    blueoil::Tensor expect = blueoil::Tensor::array({2, 3},
+						    (float *) expected_data);
+    if ((tensor.allequal(expect) == false) ||
+	(tensor.allclose(expect) == false)) {
```

Comment:
The one thing I see relatively frequently but could not get is comparing floating values against `true` and `false` constants... :thinking: 

Why not just use logical not (`!`) here?